app/controllers/reviews_controller.rb

```diff
@@ -10,6 +10,27 @@ def create
     redirect_to "/shelters/#{params[:shelter_id]}"
   end
 
+  def edit
+    @review = Review.find(params[:review_id])
+  end
+
+  def update
+    review = Review.find(params[:review_id])
+    if review.update(review_params) == false
+      flash[:notice] = "A required field is missing; please fill out all fields!"
+      redirect_to "/shelters/#{review.id}/reviews/edit"
+    elsif review.update(review_params) == true
+      redirect_to "/shelters/#{review.shelter_id}"
+    end
```

Comment:
the review_params won't let it update unless every field is filled in the form. Not sure, if we need the image to be filled in before it allows it to send (it wasn't technically in the user story). I'm going to ask the instructors, but if we don't, this will need to be revised.

Reply:
I still think we may render the flash messages in the application.html, but will take a closer look.